resources/elasticsearch/serverless.yml

```diff
@@ -61,23 +61,28 @@ resources:
             Action:
              - "es:ESHttpHead"
              - "es:ESHttpGet"
-             - "es:ESHttpPost"
             Resource: "arn:aws:es:${self:provider.region}:*:domain/${self:custom.publicDomain}/*"
+          - Effect: Allow
+            Principal: "*"
+            Action:
+             - "es:ESHttpPost"
+            Resource:
+              - "arn:aws:es:${self:provider.region}:*:domain/${self:custom.publicDomain}/*_analyze"
+              - "arn:aws:es:${self:provider.region}:*:domain/${self:custom.publicDomain}/*_cat"
+              - "arn:aws:es:${self:provider.region}:*:domain/${self:custom.publicDomain}/*_count"
+              # _field_stats and _field_caps are same https://www.elastic.co/guide/en/elasticsearch/reference/5.6/search-field-stats.html
+              # though AWS supports only the deprecated _field_stats for the moment
+              - "arn:aws:es:${self:provider.region}:*:domain/${self:custom.publicDomain}/*_field_stats"
+              - "arn:aws:es:${self:provider.region}:*:domain/${self:custom.publicDomain}/*_field_caps"
+              - "arn:aws:es:${self:provider.region}:*:domain/${self:custom.publicDomain}/*_mtermvectors"
+              - "arn:aws:es:${self:provider.region}:*:domain/${self:custom.publicDomain}/*_search"
           # Lambda can take actions on the ES domain
           - Effect: Allow
```

Comment:
Here, you basically grant all lambdas the full rights on ES. Not sure it's the best solution. Ideally, rights should be granted on a per-lambda model as discussed here: https://docs.aws.amazon.com/elasticsearch-service/latest/developerguide/es-aws-integrations.html#es-aws-integrations-s3-lambda-es (_Granting Authorization to Add Data to Your Amazon ES Domain_).

Btw, I don't know why we have this `sts:AssumeRole`. What's its purpose?

Reply:
`sts:AssumeRole` has been removed, I must have added it in the past, though it's not needed any more.